lib/DXIL/DxilOperations.cpp

```diff
@@ -2719,26 +2717,22 @@ const char *OP::GetAtomicOpName(DXIL::AtomicBinOpCode OpCode) {
 }
 
 OP::OpCodeClass OP::GetOpCodeClass(OpCode opCode) {
-  DXASSERT(0 <= (unsigned)opCode && opCode < OpCode::NumOpCodes,
-           "otherwise caller passed OOB index");
   return m_OpCodeProps[(unsigned)opCode].opCodeClass;
 }
 
 const char *OP::GetOpCodeClassName(OpCode opCode) {
-  DXASSERT(0 <= (unsigned)opCode && opCode < OpCode::NumOpCodes,
-           "otherwise caller passed OOB index");
   return m_OpCodeProps[(unsigned)opCode].pOpCodeClassName;
 }
 
 llvm::Attribute::AttrKind OP::GetMemAccessAttr(OpCode opCode) {
-  DXASSERT(0 <= (unsigned)opCode && opCode < OpCode::NumOpCodes,
-           "otherwise caller passed OOB index");
```

Comment:
Given the lack of a robust response to an invalid `opCode` like what `IsOverloadLegal` has, I don't know why removing these asserts makes sense. Perhaps it was in response to this comment? https://github.com/microsoft/DirectXShaderCompiler/pull/6302/files#r1489723490 That was in a case where an OOB opcode was handled well and there were two asserts at least one of which was clearly redundant. That's not the case here. If I'm developing the compiler and I send an invalid opcode to one of these functions, I'd rather see an assert than have an access violation that may crash immediately in the best case or just corrupt things int eh worst case.

Reply:
It being an enum satisfies me